app/notification.go

```diff
@@ -333,8 +333,8 @@ func sendNotificationEmail(post *model.Post, user *model.User, channel *model.Ch
 	if *utils.Cfg.EmailSettings.EnableEmailBatching {
 		var sendBatched bool
 		if result := <-Srv.Store.Preference().Get(user.Id, model.PREFERENCE_CATEGORY_NOTIFICATIONS, model.PREFERENCE_NAME_EMAIL_INTERVAL); result.Err != nil {
-			// if the call fails, assume it hasn't been set and don't batch notifications for this user
-			sendBatched = false
+			// if the call fails, assume that the interval has not been explicitly set and batch the notifications
+			sendBatched = true
```

Comment:
IMO the Go way of doing this would be to return an error with an `IsEmptyResult() bool` method that returns `true` for empty results.

This is done in several instances in the standard library. For example, a lot of `net` code has errors that may be "temporary":

```go
type temporary interface {
        Temporary() bool
}
 
// IsTemporary returns true if err is temporary.
func IsTemporary(err error) bool {
        te, ok := err.(temporary)
        return ok && te.Temporary()
}
```

The `os` package as the same sort of API for determining if a file error is a not-found error.